src/api/libcellml/componententity.h

```diff
@@ -47,11 +47,41 @@ class LIBCELLML_EXPORT ComponentEntity: public ImportedEntity
     ComponentEntity& operator=(ComponentEntity rhs); /**< Assignment operator */
 
     /**
-     * @brief Add a child Units to this component entity.
-     * Add a copy of the given Units u as a child units of this component entity.
-     * @param u The Units to add.
+     * @brief Add a child units to this component entity.
+     * Add a copy of the given units as a child units of this component entity.
+     * @param units The units to add.
      */
-    void addUnits(const UnitsPtr &u);
+    void addUnits(const UnitsPtr &units);
+
+    /**
+     * @brief Remove the units with the given name.
+     * Remove the first units found that matches the name @p name.  If the
+     * name is not found throw @c std::out_of_range.
+     * @param name The name of the units to remove.
+     */
+    void removeUnits(const std::string &name);
+
+    /**
+     * @brief Remove the units with the given pointer.
+     * Remove the first units found with the pointer @p units.  If the
```

Comment:
Same here (can there be more than one units with the same name?), not least because it's a pointer. (Same elsewhere in the code in fact.)